runtime/near-wallet-contract/src/lib.rs

```diff
@@ -26,18 +26,28 @@ static LOCALNET: WalletContract =
     WalletContract::new(include_bytes!("../res/wallet_contract_localnet.wasm"));
 
 /// Get wallet contract code for different Near chains.
-pub fn wallet_contract(chain_id: &str, protocol_version: ProtocolVersion) -> Arc<ContractCode> {
-    match chain_id {
-        chains::MAINNET => MAINNET.read_contract(),
-        chains::TESTNET => {
-            if protocol_version < NEW_WALLET_CONTRACT_VERSION {
-                OLD_TESTNET.read_contract()
-            } else {
-                TESTNET.read_contract()
-            }
+pub fn wallet_contract(code_hash: CryptoHash) -> Option<Arc<ContractCode>> {
+    fn check(code_hash: &CryptoHash, contract: &WalletContract) -> Option<Arc<ContractCode>> {
+        let magic_bytes = contract.magic_bytes();
+        if code_hash == magic_bytes.hash() {
+            Some(contract.read_contract())
+        } else {
+            None
         }
-        _ => LOCALNET.read_contract(),
```

Comment:
This allowed the `LOCALNET` contract to be deployed on test chains. i.e. `mocknet`